server/src/test/java/org/elasticsearch/indices/recovery/DanglingIndicesIT.java

```diff
@@ -0,0 +1,114 @@
+            .build();
+    }
+
+    /**
+     * Check that when dangling indices are discovered, then they are recovered into
+     * the cluster, so long as the recovery setting is enabled.
+     */
+    public void testDanglingIndicesAreRecoveredWhenSettingIsEnabled() throws Exception {
+        logger.info("--> starting cluster");
+        final Settings settings = buildSettings(true);
+        internalCluster().startNodes(3, settings);
+
+        // Create an index and distribute it across the 3 nodes
+        createAndPopulateIndex(INDEX_NAME, 1, 2);
+        ensureGreen();
+
+        // This is so that when then node comes back up, we have a dangling index that can be recovered.
+        logger.info("--> restarted a random node and deleting the index while it's down");
+        internalCluster().restartRandomDataNode(new InternalTestCluster.RestartCallback() {
+
+            @Override
+            public Settings onNodeStopped(String nodeName) throws Exception {
+                logger.info("--> deleting test index: {}", INDEX_NAME);
+                assertAcked(client().admin().indices().prepareDelete(INDEX_NAME));
+                return super.onNodeStopped(nodeName);
+            }
+        });
+
+        ensureGreen();
+
+        assertTrue("Expected dangling index " + INDEX_NAME + " to be recovered", indexExists(INDEX_NAME));
+    }
+
+    /**
+     * Check that when dangling indices are discovered, then they are not recovered into
+     * the cluster when the recovery setting is disabled.
+     */
+    public void testDanglingIndicesAreNotRecoveredWhenSettingIsDisabled() throws Exception {
+        logger.info("--> starting cluster");
+        internalCluster().startNodes(3, buildSettings(false));
+
+        // Create an index and distribute it across the 3 nodes
+        createAndPopulateIndex(INDEX_NAME, 1, 2);
+
+        // Create another index so that once we drop the first index, we
+        // can still assert that the cluster is green.
+        createAndPopulateIndex(INDEX_NAME + "-other", 1, 2);
+
+        ensureGreen();
+
+        // This is so that when then node comes back up, we have a dangling index that could
+        // be recovered, but shouldn't be.
+        logger.info("--> restarted a random node and deleting the index while it's down");
+        internalCluster().restartRandomDataNode(new InternalTestCluster.RestartCallback() {
+
+            @Override
+            public Settings onNodeStopped(String nodeName) throws Exception {
+                logger.info("--> deleting test index: {}", INDEX_NAME);
+                assertAcked(client().admin().indices().prepareDelete(INDEX_NAME));
+                return super.onNodeStopped(nodeName);
+            }
+        });
+
+        ensureGreen();
+
+        assertFalse("Did not expect dangling index " + INDEX_NAME + " to be recovered", indexExists(INDEX_NAME));
+    }
+
+    private void createAndPopulateIndex(String name, int shardCount, int replicaCount) {
```

Comment:
I would prefer not to have this method, just do the actual calls 

Note that there is also the `createIndex(String name, Settings indexSettings)` on ESIntegTestCase which allows to write this succinctly.